dataGeneration/generateData.js

```diff
@@ -0,0 +1,143 @@
+  var writeStream = fs.createWriteStream('./dataFiles/PaymentMethods.csv', {'flags': 'a', 'encoding': null});
+  var typeOfPayment = ['Visa', 'Master Card', 'Discover', 'UCB', 'Diners Club', 'UnionPay', 'American Express']
+  writeStream.once('open', (fd) => {
+    if (first) {
+      first = false;
+      writeStream.write('id; userId\n');
+    }
+    count = 1;
+    second = false;
+    //4000000
+    for (var i = 1; i <= 4000000; i++) {
+      var paymentId = i;
+      // var lastFourDigits = faker.helpers.replaceSymbolWithNumber("####");
+      // var type = typeOfPayment[Math.floor(Math.random() * typeOfPayment.length)];
+      var userId = count;
+
+      if (second) {
+        second = false;
+        count += 1;
+      } else {
+        second = true;
+      }
+      // REMEMBER PAYMENTID IS NOT NECCESSARILY  1-..., IT COULD BE RANDOM UNIQUE IDS
+      writeStream.write(`${paymentId}; ${userId}\n`);
+    }
+    writeStream.end();
+    console.log('done');
+  });
+}
+
+//a user makes one transactions
+//first step is to write 1 line of data that resembles my schema to a csv file.
+function generateUserTransData() {
+  var first = true;
+  var writeStream = fs.createWriteStream('./dataFiles/UserTrans.csv', {'flags': 'a', 'encoding': null});
+
+  writeStream.once('open', (fd) => {
+    if (first) {
+      first = false;
+      writeStream.write('date; userId; paymentMethodId; status; fullName; addressLine1; addressLine2; city; state; zip; country; phone; grandTotal\n');
+    } 
+    var statusArr = ['pending', 'completed', 'failed'];
+    count =- 1;
+    //2000000
+    for (var i = 1; i <= 2000000; i++) {
+      if (i % 100000 === 0) {
+        console.log(i);
+      }
+      var userTransId = i; //number from 1 - 2M
+      var date = faker.date.past().toISOString();
+
+      var userId = i; //could be random uui,
+      //insert into payment methods table userId 2 times id, id+1
+      count = count + 2;
+      var paymentMethodId = count; //each user has different paymentIds
+
+      var status = statusArr[Math.floor(Math.random() * 3)];
+      var fullName = faker.name.findName();
+      var addressLine1 = faker.fake("{{address.streetAddress}}");
+      var addressLine2 = faker.fake("{{address.secondaryAddress}}");
+      var city = faker.fake("{{address.city}}");
+      var state = faker.fake("{{address.state}}");
+      var zip = faker.fake("{{address.zipCode}}");
+      var country = faker.fake("{{address.country}}");
+      var phone = faker.fake("{{phone.phoneNumberFormat}}");
+      var grandTotal = faker.commerce.price();
+
+      //console.log(`${userTransId}, ${date}, ${userId}, ${paymentMethodId}, ${status}, ${fullName}, ${addressLine1}, ${addressLine2}, ${city}, ${state}, ${zip}, ${country}, ${phone}, ${grandTotal}\n`);
+      writeStream.write(`${date}; ${userId}; ${paymentMethodId}; ${status}; ${fullName}; ${addressLine1}; ${addressLine2}; ${city}; ${state}; ${zip}; ${country}; ${phone}; ${grandTotal}\n`);
```

Comment:
This line is really long. Would it be possible to make it two lines?

Reply:
I couldn't make it to two lines because it's using ES6's string format, so if I enter and make part of it to the next line, the actual string will have a new line in it and it won't write to my csv file correctly.

Reply:
ok sounds good. Just checking.